test/ts/integration/creditor_proxy.ts

```diff
@@ -0,0 +1,931 @@
+                let relayerBalanceBefore: BigNumber;
+                let creditorProxyBalanceBefore: BigNumber;
+
+                let receipt: Web3.TransactionReceipt;
+                let block: Web3.BlockWithoutTransactionData;
+
+                let logs: ABIDecoder.DecodedLog[];
+
+                before(async () => {
+                    await setupDebtOffer();
+                    await setupBalancesAndAllowances();
+
+                    principalToken = await DummyTokenContract.at(
+                        debtOffer.getPrincipalToken(),
+                        web3,
+                        TX_DEFAULTS,
+                    );
+
+                    [
+                        debtorBalanceBefore,
+                        creditorBalanceBefore,
+                        underwriterBalanceBefore,
+                        relayerBalanceBefore,
+                        creditorProxyBalanceBefore,
+                    ] = await getAgentBalances(principalToken);
+
+                    const txHash = await creditorProxy.fillDebtOffer.sendTransactionAsync(
+                        debtOffer.getCreditor(),
+                        debtOffer.getOrderAddresses(),
+                        debtOffer.getOrderValues(),
+                        debtOffer.getOrderBytes32(),
+                        debtOffer.getSignaturesV(),
+                        debtOffer.getSignaturesR(),
+                        debtOffer.getSignaturesS(),
+                        { from: filler },
+                    );
+
+                    receipt = await web3.eth.getTransactionReceipt(txHash);
+                    block = await web3.eth.getBlock(receipt.blockNumber);
+
+                    logs = _.compact(ABIDecoder.decodeLogs(receipt.logs));
+                });
+
+                it("should deduct principal + creditor fee from creditor account", async () => {
+                    const balance = await principalToken.balanceOf.callAsync(
+                        debtOffer.getCreditor(),
+                    );
+                    const expectedBalance = creditorBalanceBefore.minus(
+                        debtOffer.getPrincipalAmount().plus(debtOffer.getCreditorFee()),
+                    );
+                    expect(balance.toString()).to.equal(expectedBalance.toString());
+                });
+
+                it("should transfer newly minted debt token to creditor", async () => {
+                    await expect(
+                        debtTokenContract.ownerOf.callAsync(
+                            new BigNumber(debtOffer.getAgreementId()),
+                        ),
+                    ).to.eventually.equal(debtOffer.getCreditor());
+                });
+
+                it("should not change the creditor proxy balance", async () => {
+                    const balance = await principalToken.balanceOf.callAsync(creditorProxy.address);
+                    expect(balance.toString()).to.equal(creditorProxyBalanceBefore.toString());
+                });
```

Comment:
Since this is an integration test, I think it should test that the principal is transferred from the creditor to the debtor, too.

@graemecode @saturnial thoughts?

Reply:
Seems like this check would be the responsibility of the DebtKernel's integration tests since the logic that handles this isn't part of the creditor proxy. The creditor proxy's integration w other contracts obviously involves calling `fillDebtOrder` but it seems like checking that:
 - the principal is transferred from the creditor to the proxy
 - the proxy's balance doesn't change during the tx
 - the `debtOrderFilled` log is properly emitted

would catch any errors in the way `fillDebtOrder` is called by the proxy.

We'd be happy to add it though if you or @graemecode or @saturnial thinks it's be valuable.

Reply:
Oops I had added the test to check debtor balance before @bohendo posted the above comment. But I do feel this test is redundant since testing principal transfer from creditor proxy checks the amount was debited from the correct account and success of fillDebtOrder the amount was credited to the correct account.